components/camel-robotframework/src/main/java/org/apache/camel/component/robotframework/RobotFrameworkEndpoint.java

```diff
@@ -0,0 +1,149 @@
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getCriticalTags() != null ? configuration.getCriticalTags() : "").split(",")), "-c");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getNonCriticalTags() != null ? configuration.getNonCriticalTags() : "").split(",")), "-n");
+
+        // create variables from camel exchange to pass into robot
+        List<String> variables = RobotFrameworkCamelUtils.createRobotVariablesFromCamelExchange(exchange);
+        exchange.getIn().setHeader(RobotFrameworkCamelConstants.CAMEL_ROBOT_VARIABLES, variables);
+        generatedArguments.addListToArguments(variables, "-v");
+
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getVariableFiles() != null ? configuration.getVariableFiles() : "").split(",")), "-V");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getTagStatIncludes() != null ? configuration.getTagStatIncludes() : "").split(",")), "--tagstatinclude");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getTagStatExcludes() != null ? configuration.getTagStatExcludes() : "").split(",")), "--tagstatexclude");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getCombinedTagStats() != null ? configuration.getCombinedTagStats() : "").split(",")), "--tagstatcombine");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getTagDocs() != null ? configuration.getTagDocs() : "").split(",")), "--tagdoc");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getTagStatLinks() != null ? configuration.getTagStatLinks() : "").split(",")), "--tagstatlink");
+        generatedArguments.addListToArguments(Arrays.asList((configuration.getListeners() != null ? configuration.getListeners() : "").split(",")), "--listener");
+
+        // process path and set robot env by that to specify which test cases to
+        // run
+        // either from a directory or from a file
+        String path = getResourceUri();
+        ObjectHelper.notNull(path, "resourceUri");
+        log.info("RobotFrameworkEndpoint resourceUri:{}", path);
+        
+        String newResourceUri = exchange.getIn().getHeader(RobotFrameworkCamelConstants.CAMEL_ROBOT_RESOURCE_URI, String.class);
+        if (newResourceUri != null) {
+            exchange.getIn().removeHeader(RobotFrameworkCamelConstants.CAMEL_ROBOT_RESOURCE_URI);
+            log.debug("{} set to {} setting resourceUri to pass robotframework", RobotFrameworkCamelConstants.CAMEL_ROBOT_RESOURCE_URI, newResourceUri);
+            path = newResourceUri;
+        }
+
+        if (configuration.getXunitFile() == null) {
+            String testCasesFolderName = path;
+            configuration.setXunitFile(new File("TEST-" + testCasesFolderName.replace(' ', '_') + ".xml"));
```

Comment:
Can we just do like this?:
```suggestion
            configuration.setXunitFile(new File("TEST-" + path.replace(' ', '_') + ".xml"));
```
and remove `testCasesFolderName`?